pkg/cluster/executor/local.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+)
+
+// Local execute the command at local host.
+type Local struct {
+}
+
+var _ Executor = &Local{}
+
+// Execute implements Executor interface.
+func (l *Local) Execute(cmd string, sudo bool, timeout ...time.Duration) (stdout []byte, stderr []byte, err error) {
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	if len(timeout) > 0 {
+		ctx, cancel = context.WithTimeout(ctx, timeout[0])
+		defer cancel()
+	}
+
+	args := strings.Split(cmd, " ")
+	command := exec.CommandContext(ctx, args[0], args[1:]...)
+
+	stdoutBuf := new(bytes.Buffer)
+	stderrBuf := new(bytes.Buffer)
+	command.Stdout = stdoutBuf
+	command.Stderr = stderrBuf
+
+	err = command.Run()
+	stdout = stderrBuf.Bytes()
+	stderr = stderrBuf.Bytes()
+	return
+}
+
+// Transfer implements Executer interface.
+func (l *Local) Transfer(src string, dst string, download bool) error {
```

Comment:
I think we should call "cp -r"  in case of that the src and dst is a directory.

Reply:
depend on the expected behavior of `Transfer` in `Executor` interface`.
the default implementations we use `EasySSHExecutor` DO NOT support directory actually.
I will keep not support directory int this now in case of misuse.